psutil/arch/osx/cpu.c

```diff
@@ -109,7 +114,78 @@ psutil_cpu_stats(PyObject *self, PyObject *args) {
     );
 }
 
+#if defined(__arm64__) || defined(__aarch64__)
+PyObject *
+psutil_cpu_freq(PyObject *self, PyObject *args) {
+    uint32_t min;
+    uint32_t curr;
+    uint32_t pMin;
+    uint32_t eMin;
+    uint32_t max;
+    kern_return_t status;
+
+    CFDictionaryRef matching = IOServiceMatching("AppleARMIODevice");
+    if (matching == 0) {
+        return PyErr_Format(PyExc_RuntimeError, "IOServiceMatching call failed, 'AppleARMIODevice' not found");
+    }
+    io_iterator_t iter;
+
+    status = IOServiceGetMatchingServices(kIOMainPortDefault, matching, &iter);
+    if (status != KERN_SUCCESS) {
+        return PyErr_Format(PyExc_RuntimeError, "IOServiceGetMatchingServices call failed");
+    }
+
+    io_registry_entry_t entry;
+    while ((entry = IOIteratorNext(iter))) {
+        io_name_t name;
+        status = IORegistryEntryGetName(entry, name);
+        if (status != KERN_SUCCESS) {
+            continue;
+        }
+        if (strcmp(name, "pmgr") == 0) {
+            break;
+        }
```

Comment:
You still need to release entry here during the iteration before you advance to the next (in the case where you don't break).